src/Features/CSharp/Portable/ConvertSwitchStatementToExpression/ConvertSwitchStatementToExpressionCodeFixProvider.Rewriter.cs

```diff
@@ -0,0 +1,106 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.ConvertSwitchStatementToExpression
+{
+    using static SyntaxFactory;
+
+    internal sealed partial class ConvertSwitchStatementToExpressionCodeFixProvider
+    {
+        private sealed class Rewriter : CSharpSyntaxVisitor<SyntaxNode>
+        {
+            private List<ExpressionSyntax> _assignmentTargetsOpt;
+
+            private Rewriter()
+            {
+            }
+
+            public static ExpressionSyntax Rewrite(SwitchStatementSyntax node, out List<ExpressionSyntax> assignmentTargetsOpt)
+            {
+                var rewriter = new Rewriter();
+                var result = rewriter.VisitSwitchStatement(node);
+                assignmentTargetsOpt = rewriter._assignmentTargetsOpt;
+                return (ExpressionSyntax)result;
+            }
+
+            public override SyntaxNode VisitCasePatternSwitchLabel(CasePatternSwitchLabelSyntax node)
+            {
+                return node.Pattern;
+            }
+
+            public override SyntaxNode VisitCaseSwitchLabel(CaseSwitchLabelSyntax node)
+            {
+                return ConstantPattern(node.Value);
+            }
+
+            public override SyntaxNode VisitDefaultSwitchLabel(DefaultSwitchLabelSyntax node)
+            {
+                return DiscardPattern();
+            }
+
+            public override SyntaxNode VisitSwitchSection(SwitchSectionSyntax node)
+            {
+                Debug.Assert(node.Labels.Count == 1);
```

Comment:
how do you know this is true?  it looks like below that you arbitrarily recurse on a switch.  how have you validated that there is only one label per section?

Reply:
oh, i think i get it.  this should be true because hte analysis phase was pretty strict.